### ISSUE TITLE
Vulkan: Environment fog's Sun Scatter isn't fully drawn over the background sky

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#44466]
**Godot version:** `master` Git 92d4deeda

**OS/device including version:** Fedora 33, GeForce GTX 1080 (NVIDIA 455.45.01)

**Issue description:**

The Fog Sun Scatter property isn't fully drawn over the sky, which breaks the illusion of the fog smoothly fading into the sky even if Aerial Perspective is set to 1:

#### Sun Scatter: 1

![Sun Scatter: 1](https://i.imgur.com/vIjXa1t.jpg)

#### Sun Scatter: 0

![Sun Scatter: 0](https://i.imgur.com/aCik8ZI.jpg)

#### Sun Scatter: 1

![Sun Scatter: 1](https://user-images.githubusercontent.com/180032/102522405-7bba2180-4096-11eb-95f5-a7b2720875a6.png)

#### Sun Scatter: 0

![Sun Scatter: 0](https://user-images.githubusercontent.com/180032/102522501-9be9e080-4096-11eb-9fae-114077cf3444.png)

If you set Sun Scatter to a very high value, you'll notice that it *is* being drawn over the background sky, but not strongly enough.

![image](https://user-images.githubusercontent.com/180032/102523910-85dd1f80-4098-11eb-828f-a7a44e71256a.png)

If this can't be fixed, this should be documented in the class reference.

I'm using a PhysicalSkyMaterial on all the above screenshots, but I can also reproduce this issue when using ProceduralSkyMaterial.

**Steps to reproduce:**

- Create a new project.
- Add a MeshInstance with a large PlaneMesh.
- Edit `default_env.tres` in the FileSystem dock, add a Sky (PhysicalSkyMaterial or ProceduralSkyMaterial) then enable fog and sun scatter.

**Minimal reproduction project:** N/A

**Comment by clayjohn** [comment]
I'm not sure this is a bug. Sun scatter is impacted by the distance from the sample to the viewer. So it looks different on objects in the foreground vs the background. You are noticing a stark difference because your world drops off suddenly and the sky is so much further away.

**Comment by Calinou** [comment]
Closing per the above comment.

**Comment by ArtyIF** [comment]
> I'm not sure this is a bug. Sun scatter is impacted by the distance from the sample to the viewer. So it looks different on objects in the foreground vs the background. You are noticing a stark difference because your world drops off suddenly and the sky is so much further away.

Is it? I made a hemisphere with a 4000m radius/height and set up a camera with a 4000m far plane. Here's fog with Sun Scatter set to 1 with the hemisphere:

![image](https://github.com/godotengine/godot/assets/8017858/4b24c37c-255e-4240-aeab-cca7e6d71d77)

And here it is without the hemisphere:

![image](https://github.com/godotengine/godot/assets/8017858/790e4dbd-96c7-4988-8ff9-026e37367b82)

If I understood the comment correctly, the results should be equivalent.

Actually, after playing around, I realized that for whatever reason the sun doesn't affect the sky at all. I increased the Sun Scatter to 1000 and disabled the glow effect, and it doesn't affect the sky at all:

![image](https://github.com/godotengine/godot/assets/8017858/84e724ac-e960-4d7a-946d-0ff8eba57ed4)

Tested on Godot 4.1.1, Forward+ renderer, Fedora Linux 38, AMD Radeon RX 570 8GB
